runtime/android/core/src/org/xwalk/core/XWalkActivity.java

```diff
@@ -231,22 +232,37 @@ protected void dismissDialog() {
     }
 
     private void getXWalkLibrary() {
-        try {
-            Intent intent = new Intent(Intent.ACTION_VIEW);
-            startActivity(intent.setData(Uri.parse(XWALK_CORE_MARKET_URL)));
-            showDialog(getMarketProgressDialog());
-        } catch (ActivityNotFoundException e) {
-            Log.d(TAG, "Market open failed");
-            showDialog(getMarketOpenFailedDialog());
-            // TODO(sunlin): enable direct download in the future
-            // downloadXWalkLibrary();
```

Comment:
I illustrated setting methods in commit log, and will document it on website. As you see in commit log, maybe instructions around here are too long to be commented in code?